wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -89,6 +90,37 @@ public static function getSubscribedEvents() {
     ];
   }
 
+  /**
+   *
+   */
+  public static function assignVolunteerAsCampInitiator(string $op, string $objectName, $objectId, &$objectRef) {
+    if ($op !== 'edit' || $objectName !== 'AfformSubmission') {
+      return FALSE;
+    }
+
+    try {
+      $data = $objectRef['data'] ?? [];
+      if (!$data) {
+        return;
+      }
+
+      $campId = $data['Eck_Collection_Camp1'][0]['id'];
+      $volunteerId = $data['Individual6'][0]['id'];
+
+      if (empty($campId) || empty($volunteerId)) {
+        return;
+      }
+
+      EckEntity::update('Collection_Camp', FALSE)
+        ->addValue('Collection_Camp_Core_Details.Contact_Id', $volunteerId)
+        ->addWhere('id', '=', $campId)
+        ->execute();
+    }
+    catch (\Throwable $e) {
+      return FALSE;
```

Comment:
Please add some logs here. Otherwise, how will we know in the future if any error occurs?